packages/ag-charts-community/src/chart/tooltip/springAnimation.ts

```diff
@@ -0,0 +1,114 @@
+export interface SpringAnimationUpdateEvent {
+    x: number;
+    y: number;
+}
+
+const M = 0.1;
+const K = 200;
+const C = 12;
+
+const DELTA = 0.5;
+
+export class SpringAnimation {
+    private x1 = NaN;
+    private y1 = NaN;
+    private x = NaN;
+    private y = NaN;
+    private vx = 0;
+    private vy = 0;
+    private t0 = NaN;
+    private animationFrameHandle: number | undefined = undefined;
+
+    private onUpdate: ((e: SpringAnimationUpdateEvent) => void) | null = null;
+
+    addEventListener(_type: 'update', handler: (e: SpringAnimationUpdateEvent) => void) {
+        this.onUpdate = handler;
```

Comment:
This doesn't really follow the pattern for other event emitters, could we either support multiple listeners, or extend `BaseListener` to inherit all of the default behaviours and handling? Or fail by throwing an error if this would overwrite an already registered listener?